### ISSUE TITLE
Acquisition Engine has missing metadata

### ISSUE THREAD

**Issue by nanthony21** [nicost/micro-manager#17]
Images passed to the image processor plugins have the value `""` for the `Camera` metadata field. This causes line 71 of `FlipperProcessor.java` to evaluate as false meaning that the image flipper plugin doesn't no longer works during acquisitions.

**Comment by nanthony21** [comment]
Sorry I should have been more specific in the first post. Everything works fine using Live mode or the Snap button. If you take an image using the MDA you will see that the `Camera` field of the metadata is blank.

**Comment by nicost** [comment]
But not when checking the "Time Points" checkbox and setting the Count to 2 or higher...  There is surely something very strange going on here.  It may be happening in the Acquisition Engine, in case we are doomed since that is written in Clojure.

**Comment by nanthony21** [comment]
That's strange. 

This issue doesn't appear in Beta. Has the clojure changed since then?

**Comment by nicost** [comment]
No, but the way data coming from the Clojure code is handled has.  It very much seems to be a timing issue, i.e. I can force the Camera property to show up by setting breakpoints at strategic positions (for instance, in org.micromanager.acquisition.internalAcquisitionWrapperEngine.java on line 136), demonstrating that there is a thread synchronization error hidden somewhere.  The good thing is that this bug only shows up in very short acquisitions (which I never use, hence, I never noticed this).  Therefore, I am not completely sure how much effort this bug warrants.

**Comment by nanthony21** [comment]
It looks to me like it shows up in any situation where someone is using multiple positions, zstacks, or channels, regardless of what the timepoints are set to. 

I wasn't able to get it working with breakpoints like you mentioned. 

Is the fix likely somewhere in the `AcquisitionWrapperEngine` or is there a better place to start?

**Comment by nicost** [comment]
I looked a bit more, and the primary problem is in the acquisition engine.  It inserts the tag "Camera" into the metadata, and sometimes sets its value to the camera being used, and other times it leaves it empty.  2.0-gamma uses the "Camera" tag in its "check" of the metadata, and when it is empty, it uses the empty value.  However, the property "Core-Camera" always has a valid string (not sure if it will always be the actual camera, but let's hope so).  2.0-beta must have been using Core-Camera to work around this bug in the acquisition engine.  I changed the code so that it will first look at Core-Camera, and only then at "Camera".  Just pushed this fix, let me know if it works for you.
